### DISCUSSION TITLE
Checking whether or not the 3D environment is inside the polygon

### DISCUSSION THREAD

**Discussion by m6hdix** [RodrigoHamuy/react-three-map#94] [General]
Hello to all my friends, I am looking for a way to find out if the 3D environment is inside a polygon or not
The polygon is developed in another component. I have the coordinates of the same polygon. I want to calculate whether this 3D environment is inside the polygon or not. Even if a phone from the 3D environment was outside the polygon, I must receive an error. Help, friends.
![image_2024-02-22_08-30-18](https://github.com/RodrigoHamuy/react-three-map/assets/96904286/ee33a571-951b-4327-81c4-f35073e80461)
![image](https://github.com/RodrigoHamuy/react-three-map/assets/96904286/efa3fd87-5afc-43e8-bcfd-07f4e6669e7b)

**Comment by m6hdix** [comment]
Is there no one to help me?

**Comment by m6hdix** [comment]
Pless help me

**Comment by RodrigoHamuy** [comment]
@m6hdix this is not related to the library, so I'm moving to this new section "discussion > general".

I think what you want is one of these 2:

- check if a point is inside a polygon: https://www.npmjs.com/package/@turf/boolean-point-in-polygon
- Substract the difference between `model - environment`, if it returns any polygon, that's the bit outside: https://github.com/Turfjs/turf/tree/master/packages/turf-difference
